server/app/configure/authentication/local.js

```diff
@@ -53,5 +54,66 @@ module.exports = function (app) {
         passport.authenticate('local', authCb)(req, res, next);
 
     });
+    app.post('/signup', function (req, res, next) {
+
+        var authCb = function (err, user) {
+
+            if (err) return next(err);
+            if (!user) {
+                User.create(req.body)
+                .then(function(user){
+                    req.logIn(user, function (loginErr) {
+                        if (loginErr) return next(loginErr);
+                        // We respond with a response object that has user with _id and email.
+                        Cart.create({user_id: user._id, status: "active"})
```

Comment:
You also created a makeCart static in the carts model. Do you want to use that instead? What you did here makes sense to me - if you just want to keep this, then maybe remove the static in the carts model file?